src/main.rs

```diff
@@ -46,8 +48,48 @@ impl SquareOperator {
     }
 }
 
+struct SquareOperatorDeadlineContext {}
+
+impl SquareOperatorDeadlineContext {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl DeadlineContextT for SquareOperatorDeadlineContext {
+    fn get_deadline(&self, ctx: &ConditionContext) -> Duration {
+        Duration::new(1, 0)
+    }
+}
+
+struct SquareOperatorHandlerContext {
+    i: u32,
```

Comment:
I left the definition here since I wanted to spark a discussion on how state will be shared to the handlers. Notice that the `invoke_handler` method does not take a `&mut self` so the handler cannot maintain state across invocations. We need to answer the following questions:
1. How do we share the operator state with the handler? 
2. Does the handler require its own state across invocations? If yes, should we put that state in the operator state or allow `&mut self` in the invocation of the handler.

Reply:
Removed for now. We can revisit this later.